app/src/main/java/com/google/droidjump/leveldata/LevelData.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.leveldata;
+
+import android.content.res.Resources;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+public class LevelData {
+    private int baseSpeed;
+    private LinkedList<Integer> intervals;
+    private LinkedList<String> obstacleTypes;
+
+    public LevelData(Level level, Resources resources) {
+        intervals = new LinkedList<>();
+        obstacleTypes = new LinkedList<>();
+        if (level != Level.INFINITE)
+            getDataFromFile(level, resources);
+        else {
+            // For an infinite level, generation will be implemented
```

Comment:
I don't think you can do it like that. Infinite level is infinite, and you can't return infinite level data. Different mechanism should be implemented somewhere in the game that asks for a next obstacle/next obstacles/visible obstacles.

Reply:
Even if it's not infinite, for longer levels it may be better to not store the very long level in a file and have a generation point.

Reply:
I want to discuss it.

Reply:
I want to make it so that GameView doesn't depend on the type of the level (infinite / finite from file / finite generated). Therefore, I wanted to do all the implementation in one class. Now I've read about the "strategy" pattern. And learned how to implement this idea using separate classes. And will do so.